framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -959,6 +959,10 @@ protected void checkDefaultConstructor(ClassTree tree) {}
    */
   @Override
   public Void visitMethod(MethodTree tree, Void p) {
+    ClassTree enclosingClass = TreePathUtil.enclosingClass(getCurrentPath());
```

Comment:
Several subclasses override visitMethod and add additional warnings. Those warnings are still issued even if they should be skipped.  For visitClass, we made the method final and added `processClassTree(...)` that can be overridden by subclasses.  You should do the same for visitMethod.